saleor/payment/gateways/np_atobarai/api.py

```diff
@@ -0,0 +1,237 @@
+                "billed_amount": format_price(
+                    payment.captured_amount - payment_information.amount,
+                    payment_information.currency,
+                ),
+                "goods": goods,
+            }
+        ]
+    }
+
+    result, error_codes = np_request(config, "patch", "/transactions/update", json=data)
+
+    if not error_codes:
+        status = result["authori_result"]
+        transaction_id = result["np_transaction_id"]
+
+        if status == PaymentStatus.PENDING:
+            if cancel_error_codes := cancel(config, transaction_id).error_codes:
+                handle_unrecoverable_state(
+                    payment.order, "cancel", transaction_id, cancel_error_codes
+                )
+            error_messages = result["authori_hold"]
+            return errors_payment_result(error_messages)
+
+        return PaymentResult(
+            status=PaymentStatus.SUCCESS,
+        )
+
+    if PRE_FULFILLMENT_ERROR_CODE in error_codes:
+        logger.info(
+            "Fulfillment for payment with id %s was reported",
+            payment_information.graphql_payment_id,
+        )
+        return None
+
+    error_messages = get_error_messages_from_codes(
+        action=TRANSACTION_CHANGE, error_codes=error_codes
+    )
+    return errors_payment_result(error_messages)
+
+
+def reregister_transaction_for_partial_return(
+    config: ApiConfig,
+    payment: Payment,
+    payment_information: PaymentData,
+    shipping_company_code: Optional[str],
+    tracking_number: Optional[str],
+    refund_data: Optional[Dict[int, int]],
+) -> PaymentResult:
+    """Change transaction.
+
+    Use it after capturing the payment, otherwise `change_transaction`
+    is the preferred function.
+    """
+    psp_reference = payment.psp_reference
+    action = TRANSACTION_REGISTRATION
+
+    if not psp_reference:
+        return error_payment_result(
+            add_action_to_code(
+                action,
+                error_code=NO_PSP_REFERENCE,
+            )
+        )
+
+    if cancel_error_codes := cancel(config, psp_reference).error_codes:
+        error_messages = get_error_messages_from_codes(
+            action=TRANSACTION_CANCELLATION, error_codes=cancel_error_codes
+        )
+        return errors_payment_result(error_messages)
+
+    if refund_data:
+        goods = get_refunded_goods(config, refund_data, payment_information)
+    else:
+        goods = get_goods_with_discount(config, payment_information)
+
+    billed_amount = format_price(
+        payment.captured_amount - payment_information.amount,
+        payment_information.currency,
+    )
+
+    result, error_codes = register(
+        config,
+        payment_information,
+        billed_amount,
+        goods,
+    )
+
+    if not error_codes:
+        new_psp_reference = result["np_transaction_id"]
+
+        report(config, shipping_company_code, new_psp_reference, tracking_number)
```

Comment:
Isn't  it a risk of false positive here? `report` can return error status but we always return `SUCCESS` here